typed/tofieldset.go

```diff
@@ -137,7 +137,9 @@ func (v *toFieldSetWalker) visitMapItems(t *schema.Map, m value.Map) (errs Valid
 		v2 := v.prepareDescent(pe, tr)
 		v2.value = val
 		errs = append(errs, v2.toFieldSet()...)
-		if _, ok := t.FindField(key); !ok {
+		if val.IsNull() || (val.IsMap() && val.AsMap().Length() == 0) {
```

Comment:
zero length maps but not zero length lists? That's surprising?

Reply:
Actually, this is specific to struct fields with no values, it's not relevant for lists.